### PR TITLE
query: Moved away from querier params passed by context.

### PR DESCRIPTION
**Pull request by Bplotka**
This is stil weird since Queryable should be a base not per request, but this is better than context hack we had.

Signed-off-by: Bartek Plotka <bwplotka@gmail.com>

### CODE REVIEW

**Comment by Bplotka** [comment]
Since branch is merged, I can finally rebase and make proper PR

**Comment by Bplotka** [comment]
PTAL @fabxc

File: pkg/query/api/v1.go [hunk 1]
```diff
@@ -350,17 +346,18 @@ func (api *API) labelValues(r *http.Request) (interface{}, []error, *apiError) {
 	if !model.LabelNameRE.MatchString(name) {
 		return nil, nil, &apiError{errorBadData, fmt.Errorf("invalid label name: %q", name)}
 	}
+
 	var (
-		warnmtx  sync.Mutex
-		warnings []error
+		warnmtx             sync.Mutex
+		warnings            []error
```
Comment (fabxc) [inline pkg/query/api/v1.go]: That doesn't seem right – did you run `go fmt`?

Comment (Bplotka) [reply pkg/query/api/v1.go]: Seem that I forgot.. eh need to setup some commit pre hook finally

Comment (fabxc) [reply pkg/query/api/v1.go]: Or a save hook in your editor :)

Comment (Bplotka) [reply pkg/query/api/v1.go]: I can't easily.. I have to many different repos with different formatting ):

**Pull request merged**